### ISSUE TITLE
AppBar ignores Brightness and AnnotatedRegion doesn't work

### ISSUE THREAD

**Issue by feinstein** [flutter/flutter#67497]
For what I could find in the docs `AppBar` should automatically set the status bar style according to it's `brightness`. Not only it fails to do so, but this behavior isn't documented on the `brightness` parameter, only on the `SystemChrome.setSystemUIOverlayStyle` (that I could find at least).

This code shows the issue:

```dart
class MyApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Flutter Demo',
      theme: ThemeData(
        appBarTheme: AppBarTheme(
          elevation: 0,
          color: Theme.of(context).canvasColor,
          brightness: Brightness.light,
        ),
      ),
      home: Scaffold(
        appBar: AppBar(title: Text('Test Text')),
        body: Center(
          child: Text('Content'),
        ),
      ),
    );
  }
}
```

And here's what we get: 

![image](https://user-images.githubusercontent.com/8120472/95286621-3a0d2f00-083a-11eb-895a-de78eaac48c1.png)

Same thing setting the `brightness` as `Brightness.dark`:

![image](https://user-images.githubusercontent.com/8120472/95286772-97a17b80-083a-11eb-8483-2fc57338b2e5.png)

As you can see the `AppBar` `Text` is just white, no matter what the `brightness` is, also the Android Status Bar doesn't change colors and style at all. I even tried wrapping the `Scaffold` on a `Builder` as to see if it just couldn't find the `Theme` because of the `context`, but it didn't make any differences.

Adding a color scheme to the `ThemeData` does change the `Text` color, but this only shows one more time that the `AppBar`'s `brightness` isn't being used.

This is running on an Android AVD with Android API 29.

Now, further investigating this (as my main goal was to change the status bar color and icons color) I can't make `AnnotatedRegion` work:

```dart
class MyApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Flutter Demo',
      theme: ThemeData(
        appBarTheme: AppBarTheme(
          elevation: 0,
          color: Theme.of(context).canvasColor,
          brightness: Brightness.light,
        ),
      ),
      home: AnnotatedRegion(
        value: SystemUiOverlayStyle.dark.copyWith(
          statusBarColor: Colors.red,
          statusBarIconBrightness: Brightness.dark,
        ),
        child: Scaffold(
          appBar: AppBar(title: Text('Test Text')),
          body: Center(
            child: Text('Content'),
          ),
        ),
      ),
    );
  }
}
```

This is a standard `flutter create` App, nothing fancy on the Android side of things, I just changed the Dart code. I tried to set the `SystemUiOverlayStyle` in the `AnnotatedRegion` with `statusBarIconBrightness: Brightness.dark` and `statusBarIconBrightness: Brightness.light`, with no success.

Trying to use the "not recommended" `SystemChrome.setSystemUIOverlayStyle` works a bit:

```dart
class MyApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Flutter Demo',
      theme: ThemeData(
        appBarTheme: AppBarTheme(
          elevation: 0,
          color: Theme.of(context).canvasColor,
          brightness: Brightness.light,
        ),
      ),
      // home: MyHomePage(title: 'Flutter Demo Home Page'),
      home: Scaffold(
        appBar: AppBar(title: Text('Test Text')),
        body: Center(
          child: RaisedButton(
            child: Text('Content'),
            onPressed: () =>
                SystemChrome.setSystemUIOverlayStyle(SystemUiOverlayStyle.light),
          ),
        ),
      ),
    );
  }
}
```

Changing the `SystemChrome.setSystemUIOverlayStyle(SystemUiOverlayStyle.light)` to `SystemChrome.setSystemUIOverlayStyle(SystemUiOverlayStyle.dark)` makes no visual difference. But setting it to 

```dart
SystemChrome.setSystemUIOverlayStyle(SystemUiOverlayStyle.light.copyWith(statusBarColor: Colors.red))
```

Does change the Status Bar color to red, but I can't find any configuration that gets to change the Status Bar Icon color to a dark one. Bear in mind this code worked just fine on Flutter 1.20.4, but now it doesn't make any differences:

```dart
SystemChrome.setSystemUIOverlayStyle(
          SystemUiOverlayStyle.light.copyWith(
              statusBarColor: Colors.transparent,
              statusBarIconBrightness: Brightness.light),
        );
```

My experience with the `SystemUiOverlayStyle` throughout several Flutter versions have been pretty bad, sometimes it works and sometimes it doesn't, while we can't find much examples or documentations on this.

So to summarize:

* `AppBar` is ignoring its `brightness` parameter. 
* `AppBar` doesn't document how `brightness` affects the OS Status Bar (But `AnnotatedRegion` does make a reference to it.
* `AnnotatedRegion` doesn't change the Android Status Bar.
* `SystemChrome.setSystemUIOverlayStyle` partly changes the Android Status Bar.

<details>
  <summary>flutter doctor</summary>

```
Doctor summary (to see all details, run flutter doctor -v):
[√] Flutter (Channel stable, 1.22.0, on Microsoft Windows [Version 10.0.19041.508], locale pt-BR)

[√] Android toolchain - develop for Android devices (Android SDK version 29.0.3)
[√] Android Studio (version 4.0)
[√] IntelliJ IDEA Ultimate Edition (version 2019.3)
[√] VS Code (version 1.48.2)
[√] Connected device (1 available)

• No issues found!
```
</details>

**Comment by TahaTesser** [comment]
Hi @feinstein 
1,. `AppBa`r is ignoring its `brightness` parameter.
For me, it does work on API 29 emulator as well as a physical device

| Brightness.light | Brightness.dark |
| --------------- | --------------- |
![Screenshot_1602071893](https://user-images.githubusercontent.com/48603081/95328930-d2032b00-08c3-11eb-87e8-9b7ec49b1116.png) | ![Screenshot_1602071898](https://user-images.githubusercontent.com/48603081/95328936-d596b200-08c3-11eb-8b89-c89a4f2f2842.png)

<details> 
<summary>code sample</summary> 

```dart
import 'package:flutter/material.dart';
import 'package:flutter/services.dart';

void main() => runApp(MyApp());

class MyApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Flutter Demo',
      theme: ThemeData(
        appBarTheme: AppBarTheme(
          elevation: 0,
          color: Theme.of(context).canvasColor,
          brightness: Brightness.dark,
        ),
      ),
      home: Scaffold(
        appBar: AppBar(title: Text('Test Text')),
        body: Center(
          child: Text('Content'),
        ),
      ),
    );
  }
}
```

</details> 

<details> 
<summary>flutter doctor -v</summary> 

```dart
[✓] Flutter (Channel stable, 1.22.0, on Linux, locale en_US.UTF-8)
    • Flutter version 1.22.0 at /home/tahatesser/Code/flutter_stable
    • Framework revision d408d302e2 (8 days ago), 2020-09-29 11:49:17 -0700
    • Engine revision 5babba6c4d
    • Dart version 2.10.0

 
[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.2)
    • Android SDK at /home/tahatesser/Code/sdk
    • Platform android-30, build-tools 30.0.2
    • ANDROID_HOME = /home/tahatesser/Code/sdk
    • Java binary at: /home/tahatesser/Code/android-studio/jre/bin/java
    • Java version OpenJDK Runtime Environment (build
      1.8.0_242-release-1644-b3-6222593)
    • All Android licenses accepted.

[✓] Android Studio (version 4.0)
    • Android Studio at /home/tahatesser/Code/android-studio
    • Flutter plugin version 50.0.1
    • Dart plugin version 193.7547
    • Java version OpenJDK Runtime Environment (build
      1.8.0_242-release-1644-b3-6222593)

[✓] VS Code (version 1.49.3)
    • VS Code at /usr/share/code
    • Flutter extension version 3.15.0

[✓] Connected device (2 available)
    • RMX2001 (mobile)                   • EUYTFEUSQSRGDA6D • android-arm64 •
      Android 10 (API 29)
    • Android SDK built for x86 (mobile) • emulator-5554    • android-x86   •
      Android 10 (API 29) (emulator)

• No issues found!

```

</details> 

However other points 2..4 you mentioned are totally reproducible API 29 emulator

**Comment by feinstein** [comment]
I double checked and on a physical device `AppBar` does work, but not on my Emulator (both at Android 10 API 29).

Our `flutter doctor -v` are quite similar, could you test this on Windows? Do you have an idea why would the Emulator fail to work but on a real device this works just fine? I am worried that maybe this is a hard to track bug and a small percentage of my users will have this issue and others won't, as this appears to work on some devices and not on others.

<details>
  <summary>flutter doctor -v</summary>

```
[√] Flutter (Channel stable, 1.22.0, on Microsoft Windows [Version 10.0.19041.508], locale pt-BR)
    • Flutter version 1.22.0 at C:\Programming\Flutter\sdk\flutter
    • Framework revision d408d302e2 (8 days ago), 2020-09-29 11:49:17 -0700
    • Engine revision 5babba6c4d
    • Dart version 2.10.0


[√] Android toolchain - develop for Android devices (Android SDK version 30.0.2)
    • Android SDK at C:\Users\Michel\AppData\Local\Android\sdk
    • Platform android-30, build-tools 30.0.2
    • Java binary at: C:\Program Files\Android\Android Studio\jre\bin\java
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b01)
    • All Android licenses accepted.

[√] Android Studio (version 4.0)
    • Android Studio at C:\Program Files\Android\Android Studio
    • Flutter plugin version 50.0.1
    • Dart plugin version 193.7547
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b01)

[√] IntelliJ IDEA Ultimate Edition (version 2019.3)
    • IntelliJ at C:\Program Files\JetBrains\IntelliJ IDEA 2019.2.2
    • Flutter plugin version 44.0.3
    • Dart plugin version 193.6911.31

[√] VS Code (version 1.48.2)
    • VS Code at C:\Users\Michel\AppData\Local\Programs\Microsoft VS Code
    • Flutter extension version 3.7.1

[√] Connected device (2 available)
    • ONEPLUS A5000 (mobile)             • bc18175a      • android-arm64 • Android 10 (API 29)
    • Android SDK built for x86 (mobile) • emulator-5554 • android-x86   • Android 10 (API 29) (emulator)

• No issues found!
```
</details>

Also, shouldn't this get a `Regression` label as things that were working aren't anymore?

**Comment by TahaTesser** [comment]
Just tried oo windows running API 29 emulator, 1. case  does work 

<details> 
<summary>complete code sample</summary> 

```dart
import 'package:flutter/material.dart';
import 'package:flutter/services.dart';

void main() => runApp(MyApp());

class MyApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Flutter Demo',
      theme: ThemeData(
        appBarTheme: AppBarTheme(
          elevation: 0,
          color: Theme.of(context).canvasColor,
          brightness: Brightness.dark,
        ),
      ),
      home: Scaffold(
        appBar: AppBar(title: Text('Test Text')),
        body: Center(
          child: Text('Content'),
        ),
      ),
    );
  }
}

```

</details> 

<details> 
<summary>flutter doctor -v</summary> 

```bash
[✓] Flutter (Channel stable, 1.22.0, on Microsoft Windows [Version 10.0.19041.508], locale en-US)
    • Flutter version 1.22.0 at C:\Code\flutter_stable
    • Framework revision d408d302e2 (9 days ago), 2020-09-29 11:49:17 -0700
    • Engine revision 5babba6c4d
    • Dart version 2.10.0

[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.2)
    • Android SDK at C:\Code\sdk
    • Platform android-30, build-tools 30.0.2
    • ANDROID_HOME = C:\Code\sdk
    • Java binary at: C:\Code\android-studio\jre\bin\java
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b01)
    • All Android licenses accepted.

[✓] Android Studio (version 4.0)
    • Android Studio at C:\Code\android-studio
    • Flutter plugin version 50.0.1
    • Dart plugin version 193.7547
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b01)

[✓] VS Code (version 1.49.3)
    • VS Code at C:\Users\Taha\AppData\Local\Programs\Microsoft VS Code
    • Flutter extension version 3.15.0

[✓] Connected device (1 available)
    • Android SDK built for x86 (mobile) • emulator-5554 • android-x86 • Android 10 (API 29) (emulator)

• No issues found!
```

</details>

**Comment by feinstein** [comment]
Are you using an AVD with Play Services?

**Comment by TahaTesser** [comment]
@feinstein 
Does this count?
![image](https://user-images.githubusercontent.com/48603081/95474767-25987600-09a3-11eb-839c-579e15bd21b4.png)

**Comment by feinstein** [comment]
Yep... It's the same as mine... I have no ideas left then... Could you upload your project to git so I can run it here? Maybe it's something on the android code side. If possible also put the APK in there so I can see if it's related to the compilation.

**Comment by TahaTesser** [comment]
@feinstein 
Sure, here you go
**Video Preview**
[video.zip](https://github.com/flutter/flutter/files/5348962/video.zip)
**Project File**
[stable_flutter.zip](https://github.com/flutter/flutter/files/5348965/stable_flutter.zip)
**APK**
[app-release.zip](https://github.com/flutter/flutter/files/5348966/app-release.zip)

**Comment by feinstein** [comment]
@TahaTesser thanks. I can't see any differences, things still don't work on my AVD.

Also the AVD refused to install the APK.

**Comment by HansMuller** [comment]
I've also confirmed that AppBar uses its brightness parameter, or the AppBarTheme brightness property, as expected (https://github.com/flutter/flutter/issues/67497#issuecomment-704892459) on an Android pixel.

@feinstein - it would help to clarify the title and description of this issue. It seems to be problem with some (one?) emulator.

**Comment by feinstein** [comment]
@HansMuller as I was converting my App to use `ColorScheme` I undertood how the `AppBar` changes it's system icons brightness and so I was able to fix it.

In a nutshell, I was trying to set the system icons brightness with `AnnotatedRegion` and `SystemChrome.setSystemUIOverlayStyle` but it was competing with the `AppBar` itself, so at the end they produced no effect at all.

Since this wasn't documented I didn't realize there would be a interference between those components.  `AnnotatedRegion` and `SystemChrome.setSystemUIOverlayStyle` try to set a new brightness, but then the `AppBar` will set it too later on the same (IIRC) build cycle.

I removed the `AnnotatedRegion` and `SystemChrome.setSystemUIOverlayStyle` from my App and the brightness is working, but trying to use them both together is impossible AFAIK.

**Comment by HansMuller** [comment]
@feinstein - Just now, I'm updating AppBar to use defaults based on (just) the theme's ColorScheme and textTheme - https://github.com/flutter/flutter/pull/69251.

You're right about AppBar and its `AnnotatedRegion`. I was hoping that we didn't need to explain that apps should not also create an AnnotatedRegion for this purpose; maybe we should?

**Comment by feinstein** [comment]
### TL;DR;

I think we should, but just some brief words about it, like: 

>"You shouldn't use [AnnotatedRegion]  and [AppBar] together, as [AppBar] automatically configures the system status bar brightness and thus will revert any changes made by the [AnnotatedRegion]."

### My understanding of the problem:

Many apps (like mine) have some screens with an `AppBar` (regular screens) and some without (login screen, announcement screen, on-boarding, etc). Some screens need light system icons and some dark. Controlling the system icons brightness will be a very different mechanism from some screens than others and this leads to confusion.

If you try to google for "how to change status bar color on Flutter" you will find old code samples with  `SystemChrome.setSystemUIOverlayStyle`. If a developer starts his App with a login screen that doesn't have an `AppBar` he might not know the `AppBar` has an automatic mechanism for configuring the status bar for him. Later when he makes a screen with an `AppBar` he won't understand what's going on.

Things are a bit scarce on the docs on how to configure the status bar, specially without an AppBar. Also last time I checked the docs aren't much clear on how `AnnotatedRegion` works, or why we should avoid `SystemChrome.setSystemUIOverlayStyle`, they are very generic and brief. If any Apps need to configure the system icons brightness, without having an `AppBar` there to do things automatically for him, the developer starts feel lost on the docs.

Honestly I don't know how many devs will actually come across with this issue, I just found it because I tried to make a general system icon brightness control in the app state, changing the value being passed to `AnnotatedRegion`. My idea was to surround the whole App with an `AnnotatedRegion` and if any screen wanted to change its brightness, it just had to call the `AppState` and request it. It sounded simpler than making an `AnnotatedRegion` for each screen that doesn't have an `AppBar`, but it didn't work and later I found out why (later I also decided it wasn't a good idea and having the configuration on each page was preferable, as it's clearer what's going on on that screen). How many people will have an idea like this? Not many I suppose, but I still like to have all the mechanisms properly documented.

Maybe the `AnnotatedRegion` docs that should be improved, as there's where devs might find the corner cases?

**Comment by amitkot** [comment]
@HansMuller 
> You're right about AppBar and its `AnnotatedRegion`. I was hoping that we didn't need to explain that apps should not also create an AnnotatedRegion for this purpose; maybe we should?

In my case, I have an `AppBar` (actually, a `SliverAppBar`) where I override the `backgroundColor` to `theme.colorScheme.surface` to get a "transparent" AppBar effect.

### What I wanted
An effect similar to Messages and Calendar apps:

<img width="411" alt="Screen Shot 2020-11-01 at 17 10 41" src="https://user-images.githubusercontent.com/654041/97806709-b2aeb200-1c65-11eb-9933-364b5db43b70.png">

### Code I used:
```dart
                SliverAppBar(
                  brightness: theme.brightness,
                  pinned: true,
                  backgroundColor: theme.colorScheme.surface,
                  iconTheme: theme.iconTheme
                      .copyWith(color: theme.colorScheme.onSurface),
                  ...
               )
```
### It works on iOS:

<img width="411" alt="Screen Shot 2020-11-01 at 17 17 21" src="https://user-images.githubusercontent.com/654041/97806817-34064480-1c66-11eb-872d-3ed9895c37c8.png">

### But on Android
The status bar has a dark overlay:

<img width="411" alt="Screen Shot 2020-11-01 at 17 19 53" src="https://user-images.githubusercontent.com/654041/97806873-7f205780-1c66-11eb-9ded-5df976262a41.png">

Nothing I tried removes the dark overlay from the status bar, including wrapping the Scaffold in an `AnnotatedRegion` and even changing this page to be a `StatefulWidget` and calling `SystemChrome.setSystemUIOverlayStyle(...)` in its `initState` method.

- Using Flutter stable, 1.22.2
- I tried it with a simpler layout, with Just a Scaffold and an AppBar and it doesn't work there too.
- I get the same results on an Android device running Android 10, and on an Emulator running Android 10.

**Comment by feinstein** [comment]
You could just use `Colors.transparent`.

**Comment by amitkot** [comment]
@feinstein instead of `theme.colorScheme.surface`? 
If so, I can but the dark overlay over the status bar remains, though.

**Comment by feinstein** [comment]
Yes, I just meant you don't have to force a color when you want transparency.

**Comment by feinstein** [comment]
I am at my phone now, so I going to share with you somethings that I made in my app and I have a similar effect to what you want:

1. I create the App's `ThemeData` with `ThemeData.from` passing a `ColorScheme`. This will setup some defaults.

2. I have a a general theme for all my `AppBar`:

```dart
appBarTheme: AppBarTheme(
            elevation: 0,
            color: Colors.white,
            brightness: Brightness.light,
          ),
```

3. When I changed the language of my device, while testing Localizations, the `AppBar` started to work on the emulator. Don't ask me why.

This issue took me a lot of time to figure out what was happening. What helped me was to setup a breakpoint in the `AppBar` code, when it calls the system to change the status bar. Then I could see what were the parameters being called. I also placed a breakpoint on the system calls so I could see if other parts of my code were calling it, and in which order. Not a simple debug, but it could give you some light.

I am afraid that's all I can help for now.

**Comment by amitkot** [comment]
@feinstein I followed AppBar's `build()` function, and from what I see now, it indeed always wraps the widget it creates in an `AnnotatedRegion<SystemUiOverlayStyle>` widget which uses either `SystemUiOverlayStyle.light` or `SystemUiOverlayStyle.dark` according to the brightness provided (directly, or as part of the theme).

Is there a way to override AppBar's AnnotatedRegion?

**Comment by feinstein** [comment]
Not that I know off (I don't have the source code in front of me, but I think I remember that part).

I hope this would be addressed once this issue is closed, but @HansMuller could provide more info on that.

**Comment by amitkot** [comment]
@feinstein thanks!
Can the steps you describe [above](https://github.com/flutter/flutter/issues/67497#issuecomment-720137606) serve as a workaround for this issue? It is unclear to me if they helped.

**Comment by feinstein** [comment]
My splash screen has an `AnnotatedRegion` with:

```dart
final kSystemLightUiOverlayStyle =
    SystemUiOverlayStyle.light.copyWith(statusBarColor: Colors.transparent);
```

I don't remember how the `AppBar` sets the status bar colors, maybe it leaves the old color I set? As you can see I set it to transparent.

Try to set its color once to transparent and see if it propagates.

**Comment by amitkot** [comment]
@feinstein from what I see in this snippet, you create a copy of `SystemUiOverlayStyle.light` with a transparent color, so it should not propagate. This probably works for your splash screen as it has no `AppBar` to override it.

@HansMuller as AppBar overrides the `AnnotatedRegion`, what is the recommended way to change the status bar color to transparent when using an `AppBar`?

**Comment by feinstein** [comment]
@amitkot `AppBar` does surround it's contents into an `AnnotatedRegion`, but if you look into the `SystemUiOverlayStyle.light` source code you will see that status bar has `null` for it's color. This means the `AnnotatedRegion` inside the `AppBar` won't change the color, if you managed to set one.

As I said before you can set your status bar color to transparent and it should be respected afterwards. I don't like this behavior, I think it feels more like a hack and I can't see it documented anywhere in the `AppBar` docs. IMO if the `AppBar`  is going to automatically change the status bar, then all of it's behavior should be properly documented.

I think that the trick is to be able to set the status bar color once and not have the `AppBar` override it on the same build pass.

This code will show you how I managed to do it on my App:

```dart
void main() => runApp(MyApp());

class MyApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Flutter Demo',
      theme: ThemeData.from(
          colorScheme: ColorScheme.light(
            primary: Colors.white,
          )).copyWith(
        appBarTheme: AppBarTheme(
          elevation: 0,
          color: Colors.white,
          brightness: Brightness.light,
        ),
      ),
      home: TestApp(),
    );
  }
}

final kSystemLightUiOverlayStyle =
SystemUiOverlayStyle.light.copyWith(statusBarColor: Colors.transparent);

final kSystemDarkUiOverlayStyle =
SystemUiOverlayStyle.dark.copyWith(statusBarColor: Colors.transparent);

class TestApp extends StatelessWidget {
  Widget build(BuildContext context) {
    return AnnotatedRegion(
      value: kSystemDarkUiOverlayStyle,
      child: Scaffold(
        body: Column(
          mainAxisAlignment: MainAxisAlignment.center,
          crossAxisAlignment: CrossAxisAlignment.stretch,
          children: [
            RaisedButton(onPressed: () {
              Navigator.push(context, MaterialPageRoute(builder: (_) => AppBarPage()));
            }, child: Text('Next Screen'),),
          ],
        ),
      ),
    );
  }
}

class AppBarPage extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(title: Text('AppBar Page')),
      body: Center(
        child: Text('The AppBar should have a transparent color'),
      ),
    );
  }
}
```

**Comment by amitkot** [comment]
@feinstein so this is very interesting. Your example code works - the status bar color is indeed transparent.
I played with permutations of this code, and my conclusion is:

#### To get the transparent status bar color we have to set the `AnnotatedRegion` on the page *before* the one with the `AppBar`. 

I tried wrapping the Scaffold but it didn't work, I even tried wrapping AppBarPage inside the `MaterialPageRoute` builder but it did not work.
The only thing that I could see work is setting the `AnnotatedRegion` in the page before. 

Is this your conclusion as well or did I miss something?

**Comment by feinstein** [comment]
Yes, it is, that's why I said:

>I think that the trick is to be able to set the status bar color once and not have the AppBar override it on the same build pass. 

But you could do it without the `AnnotatedRegion`, if you don't have a page without an `AppBar`. Just call `SystemChrome.setSystemUIOverlayStyle` on `main` or something similar, I guess it should work.

**Comment by amitkot** [comment]
@feinstein I ended up indeed adding the `AnnotatedRegion` to the top widget of the app. It is responsible for preparing the providers and showing the actual widget according to user log-in state, so it has no UI of its own.

BTW, using `Colors.transparent` for the `AppBar` background did not end up working for me as the scaffold body is scrollable and so it scrolled underneath the AppBar and was visible, which is not the look I was going for. Using `Theme.of(context).colorScheme.surface` works but in combination with the transparent status bar the status bar white font became unreadable. 
I ended up using [flutter_statusbar_manager](https://pub.dev/packages/flutter_statusbar_manager) to set the brightness to dark in `initState()` and back to light in `dispose()`.

**Comment by feinstein** [comment]
Ok, I hope things are better/easier to configure when a fix comes out.

**Comment by HansMuller** [comment]
I've just landed the updated AppBar (more updates to come). I included a brief note about not using AnnotatedRegion: https://github.com/flutter/flutter/pull/69251/files#diff-20c19d254b0a82027625f1246ba51cb2264bfabf105fe91d532c19e6a6d77478R410

The comment is per https://github.com/flutter/flutter/issues/67497#issuecomment-718296684

**Comment by feinstein** [comment]
@HansMuller that comment is very useful.

I would like to propose a simpler mechanism for configuring the `SystemUiOverlayStyle`. Maybe we can have an optional `SystemUiOverlayStyle` passed to the `AppBar` and it will set whatever is passed (if passed)? Or even we could add it to the `appBarTheme` and have a default `light` and `dark` defined for the App's `Theme`. Or maybe add it to the `Theme` itself so it can be configured for parts of the App that don't have an `AppBar` as well.

I think this simple mechanism would cover most use cases.... thoughts? Should I open a new issue with this?

**Comment by HansMuller** [comment]
I agree, although I haven't tried it yet. It would be much more flexible and straightforward to just have a SystemUIOverlayStyle parameter; rather than implying the SystemUIOverlayStyle you want with the (otherwise unused) AppBar brightness parameter.
